torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -1636,15 +1636,15 @@ def _linear_f16_act_floatx_weight_check(input_tensor, weight_tensor, bias):
         )
     )
 
-def _linear_f16_act_floatx_weight_impl(input_tensor, weight_tensor, bias):
+def _linear_f16_bf16_act_floatx_weight_impl(input_tensor, weight_tensor, bias):
     from torchao.dtypes.floatx import _SPLIT_K_MAP
     from torchao.ops import quant_llm_linear
 
     act = input_tensor
     weight = weight_tensor
 
     out_dim, in_dim = weight.shape
-    act_reshaped = act.view(-1, in_dim).half()
+    act_reshaped = act.view(-1, in_dim)
```

Comment:
no need for a cast anymore?

Reply:
Is `input_tensor` allowed to be something other than float16 or bfloat16? I assumed it wasn't, so I thought that the input is always either float16 or bfloat16, making casting unnecessary. If we do cast here, how we would know whether to cast to float16 or bfloat16?